### PR TITLE
Add ClusterIssuer test

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR adds a test to check if the ClusterIssuers installed by cert-manager-app will reach Ready state

CAPZ test failure is expected. CAPV test failure seems unrelated

### CODE REVIEW

**Review by AverageMarcus — CHANGES_REQUESTED** [review]
> CAPZ test failure is expected. 

How are we fixing it? We can't merge this in without it passing or we'll block other people working.

**Comment by ubergesundheit** [comment]
> How are we fixing it? 

Any way to use this cluster-test-suites version in https://github.com/giantswarm/default-apps-azure/pull/150 ?

File: common/certmanager.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		c := condition.(map[string]interface{})
+		conditionType := c["type"]
+		status := c["status"]
+		if conditionType == "Ready" && status == "True" {
+			logger.Log("Found status.condition with type '%s' and status '%s' in ClusterIssuer '%s'", conditionType, status, clusterIssuerName)
+			return nil
+		}
+	}
+
+	return fmt.Errorf("ClusterIssuer '%s' is not Ready", clusterIssuerName)
```
Comment (AverageMarcus) [inline common/certmanager.go:81]: To give some progress update while the test is running: 
```suggestion
    logger.Log("ClusterIssuer '%s' is not Ready", clusterIssuerName)
	return fmt.Errorf("ClusterIssuer '%s' is not Ready", clusterIssuerName)
```

File: common/certmanager.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		c := condition.(map[string]interface{})
+		conditionType := c["type"]
+		status := c["status"]
+		if conditionType == "Ready" && status == "True" {
+			logger.Log("Found status.condition with type '%s' and status '%s' in ClusterIssuer '%s'", conditionType, status, clusterIssuerName)
+			return nil
+		}
+	}
+
+    logger.Log("ClusterIssuer '%s' is not Ready", clusterIssuerName)
```
Comment (ubergesundheit) [inline common/certmanager.go:81]:
```suggestion
		logger.Log("ClusterIssuer '%s' is not Ready", clusterIssuerName)
```

**Comment by AverageMarcus** [comment]
Not in CI 😞  You can when running locally by setting the `E2E_OVERRIDE_VERSIONS`.

Easiest option would be to get that PR merged in and released then once the new version is available you can re-run the tests on this PR to ensure they're passing.

File: common/certmanager.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+		c := condition.(map[string]interface{})
+		conditionType := c["type"]
+		status := c["status"]
+		if conditionType == "Ready" && status == "True" {
+			logger.Log("Found status.condition with type '%s' and status '%s' in ClusterIssuer '%s'", conditionType, status, clusterIssuerName)
+			return nil
+		}
+	}
+
+		logger.Log("ClusterIssuer '%s' is not Ready", clusterIssuerName)
```
Comment (ubergesundheit) [inline common/certmanager.go:81]:
```suggestion
	logger.Log("ClusterIssuer '%s' is not Ready", clusterIssuerName)
```

**Comment by ubergesundheit** [comment]
> Not in CI 😞 You can when running locally by setting the `E2E_OVERRIDE_VERSIONS`.
> 
> Easiest option would be to get that PR merged in and released then once the new version is available you can re-run the tests on this PR to ensure they're passing.

ok fair enough :+1:

**Review by AverageMarcus — APPROVED** [review]

**Pull request merged**